internal/functions/deploy/deploy.go

```diff
@@ -97,7 +97,15 @@ func GetFunctionConfig(slugs []string, importMapPath string, noVerifyJWT *bool,
 		// Precedence order: flag > config > fallback
 		functionDir := filepath.Join(utils.FunctionsDir, name)
 		if len(function.Entrypoint) == 0 {
-			function.Entrypoint = filepath.Join(functionDir, "index.ts")
+			indexEntrypoint := filepath.Join(functionDir, "index.ts")
+			mainEntrypoint := filepath.Join(functionDir, "main.ts")
+			if _, err := fsys.Stat(indexEntrypoint); err == nil {
+				function.Entrypoint = indexEntrypoint
+			} else if _, err := fsys.Stat(mainEntrypoint); err == nil {
+				function.Entrypoint = mainEntrypoint
+			} else {
+				return nil, errors.Errorf("Cannot find a valid entrypoint file (index.ts or main.ts) for the '%s' function. Set the custom entrypoint path in config.toml", name)
+			}
```

Comment:
```suggestion
			function.Entrypoint = filepath.Join(functionDir, "index.ts")
			if _, err := fs.Stat(fsys, function.Entrypoint); errors.Is(err, os.ErrNotExist) {
				mainEntrypoint := filepath.Join(functionDir, "main.ts")
				if _, err := fs.Stat(fsys, mainEntrypoint); err == nil {
					function.Entrypoint = mainEntrypoint
				}
			}
```

ditto

One way to reduce duplication is to add a `function.ResolvePaths` method. I can work on that separately.